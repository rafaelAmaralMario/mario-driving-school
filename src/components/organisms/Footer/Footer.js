import React from "react";
import styled from 'styled-components';
import { FaPhone, FaBuilding, FaFacebookF, FaLinkedinIn, FaTwitter } from 'react-icons/fa';

import { Container, Grid, Heading } from 'components/atoms';

export const Root = styled.footer`
  background-color: #2a2a2a;
  color: #aaa;
  padding: 40px 0;

  h6 {
    color: #fff;
    margin-top: 0;
    text-transform: uppercase;
  }
`;

export const IconContainer = styled.span`
  margin-right: 8px;
`;

export const FooterLink = styled.a`
  display: flex;
  align-items: center;
  color: ${prop => prop.theme.colors.primary.main};
  text-decoration: none;

  &:hover {
    color: ${prop => prop.theme.colors.primary.dark};
  }  
`;

const Footer = () => (
  <Root>
    <Container>
      <Grid md={3} >
        <div>
          <Heading>
            <h6>Mario Driving School</h6>
          </Heading>
          <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium eos adipisci consequatur, maiores maxime mollitia amet animi necessitatibus illum iusto voluptas numquam nam quisquam quis. Assumenda soluta laboriosam excepturi omnis?</p>
        </div>
        <div>
          <Heading>
            <h6>Contact</h6>
          </Heading>
          <p>
            <IconContainer>
              <FaPhone />
            </IconContainer>
            (12) 12345-6789
          </p>
          <p>
            <IconContainer>
              <FaBuilding />
            </IconContainer>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium eos adipisci consequatur, maiores maxime mollitia amet animi necessitatibus illum iusto voluptas numquam nam quisquam quis.
          </p>
        </div>
        <div>
          <Heading>
            <h6>Social Medias</h6>
          </Heading>
          <p>
            <FooterLink target="_blank" href="https://www.facebook.com">
              <IconContainer>
                <FaFacebookF />
              </IconContainer>
              Facebook
            </FooterLink>
          </p>
          <p>
            <FooterLink target="_blank" href="https://twitter.com">
              <IconContainer>
                <FaLinkedinIn />
              </IconContainer>
              Linkedin
            </FooterLink>
          </p>
          <p>
            <FooterLink target="_blank" href="https://linkedin.com">
              <IconContainer>
                <FaTwitter />
              </IconContainer>
              Twitter
            </FooterLink>
          </p>
        </div>
      </Grid>
    </Container>
  </Root>
);

export default Footer;
