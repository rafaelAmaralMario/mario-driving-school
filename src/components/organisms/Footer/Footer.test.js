import React from "react";

import { render } from "utils/test-utils";
import Footer from "./Footer";

test("Should render with text", () => {
  const { asFragment } = render(<Footer></Footer>)

  expect(asFragment()).toMatchSnapshot();

});
