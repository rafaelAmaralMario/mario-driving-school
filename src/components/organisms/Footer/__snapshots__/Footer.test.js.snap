// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render with text 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  padding: 0 8px;
}

.c2 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c3 h1,
.c3 h2,
.c3 h3,
.c3 h4,
.c3 h5,
.c3 h6 {
  position: relative;
  margin-bottom: 25px;
  padding-bottom: 8px;
  border-bottom: 1px solid rgba(255,255,255,0.2);
  margin-bottom: 0;
}

.c3 h1::after,
.c3 h2::after,
.c3 h3::after,
.c3 h4::after,
.c3 h5::after,
.c3 h6::after {
  content: "";
  position: absolute;
  left: 0;
  bottom: -3px;
  background-color: #ffc107;
  height: 5px;
  width: 70px;
}

.c3 h1 {
  padding-bottom: 25px;
}

.c0 {
  background-color: #2a2a2a;
  color: #aaa;
  padding: 40px 0;
}

.c0 h6 {
  color: #fff;
  margin-top: 0;
  text-transform: uppercase;
}

.c4 {
  margin-right: 8px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #ffc107;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  color: #c79100;
}

@media (min-width:undefinedpx) {
  .c1 {
    padding: 0 16px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 1140px;
    margin: 0 auto;
    padding: 0;
  }
}

@media (min-width:768px) {
  .c2 {
    grid-template-columns: repeat(3,1fr);
  }
}

<footer
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div>
          <div
            class="c3"
          >
            <h6>
              Mario Driving School
            </h6>
          </div>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium eos adipisci consequatur, maiores maxime mollitia amet animi necessitatibus illum iusto voluptas numquam nam quisquam quis. Assumenda soluta laboriosam excepturi omnis?
          </p>
        </div>
        <div>
          <div
            class="c3"
          >
            <h6>
              Contact
            </h6>
          </div>
          <p>
            <span
              class="c4"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                />
              </svg>
            </span>
            (12) 12345-6789
          </p>
          <p>
            <span
              class="c4"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M436 480h-20V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v456H12c-6.627 0-12 5.373-12 12v20h448v-20c0-6.627-5.373-12-12-12zM128 76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76zm0 96c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40zm52 148h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12zm76 160h-64v-84c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v84zm64-172c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40z"
                />
              </svg>
            </span>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium eos adipisci consequatur, maiores maxime mollitia amet animi necessitatibus illum iusto voluptas numquam nam quisquam quis.
          </p>
        </div>
        <div>
          <div
            class="c3"
          >
            <h6>
              Social Medias
            </h6>
          </div>
          <p>
            <a
              class="c5"
              href="https://www.facebook.com"
              target="_blank"
            >
              <span
                class="c4"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 320 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                  />
                </svg>
              </span>
              Facebook
            </a>
          </p>
          <p>
            <a
              class="c5"
              href="https://twitter.com"
              target="_blank"
            >
              <span
                class="c4"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                  />
                </svg>
              </span>
              Linkedin
            </a>
          </p>
          <p>
            <a
              class="c5"
              href="https://linkedin.com"
              target="_blank"
            >
              <span
                class="c4"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  />
                </svg>
              </span>
              Twitter
            </a>
          </p>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
